eth_dummy_phy/examples/simple/main/emac2emac.c

```diff
@@ -0,0 +1,297 @@
+    esp32_emac_config.smi_mdc_gpio_num = -1;
+    esp32_emac_config.smi_mdio_gpio_num = -1;
+#if CONFIG_EXAMPLE_RMII_CLK_SOURCE_DEV
+    esp32_emac_config.clock_config.rmii.clock_mode = EMAC_CLK_OUT;
+    esp32_emac_config.clock_config.rmii.clock_gpio = EMAC_CLK_OUT_180_GPIO;
+#else
+    esp32_emac_config.clock_config.rmii.clock_mode = EMAC_CLK_EXT_IN;
+    esp32_emac_config.clock_config.rmii.clock_gpio = EMAC_CLK_IN_GPIO;
+#endif // CONFIG_EXAMPLE_RMII_CLK_SOURCE_DEV
+
+    // Create new ESP32 Ethernet MAC instance
+    esp_eth_mac_t *mac = esp_eth_mac_new_esp32(&esp32_emac_config, &mac_config);
+    // Create dummy PHY instance
+    esp_eth_phy_t *phy = esp_eth_phy_new_dummy(&phy_config);
+
+    // Init Ethernet driver to default and install it
+    esp_eth_handle_t eth_handle = NULL;
+    esp_eth_config_t config = ETH_DEFAULT_CONFIG(mac, phy);
+#if CONFIG_EXAMPLE_RMII_CLK_SINK_DEV
+    // REF RMII CLK sink device performs multiple EMAC init attempts since RMII CLK source device may not be ready yet
+    int i;
+    for (i = 1; i <= 5; i++) {
+        ESP_LOGI(TAG, "Ethernet driver install attempt: %i", i);
+        if (esp_eth_driver_install(&config, &eth_handle) == ESP_OK) {
+            break;
+        }
+        vTaskDelay(pdMS_TO_TICKS(100));
+    }
+    if (i > 5) {
+        ESP_LOGE(TAG, "Ethernet driver install failed");
+        abort();
+        return;
+    }
+#else
+    ESP_ERROR_CHECK(esp_eth_driver_install(&config, &eth_handle));
+#endif // CONFIG_EXAMPLE_RMII_CLK_SOURCE_DEV
+
+    // Initialize TCP/IP network interface aka the esp-netif
+    ESP_ERROR_CHECK(esp_netif_init());
+    // Create default event loop that running in background
+    ESP_ERROR_CHECK(esp_event_loop_create_default());
+
+    // Create instance of esp-netif for Ethernet
+    esp_netif_inherent_config_t esp_netif_config = ESP_NETIF_INHERENT_DEFAULT_ETH();
+#if CONFIG_EXAMPLE_DHCP_SERVER_EN
+    esp_netif_config.flags = (esp_netif_flags_t)(ESP_NETIF_IPV4_ONLY_FLAGS(ESP_NETIF_DHCP_SERVER));
+    esp_netif_config.ip_info = &_g_esp_netif_soft_ap_ip; // Use the same IP ranges as IDF's soft AP
+    esp_netif_config.get_ip_event = 0;
+    esp_netif_config.lost_ip_event = 0;
+#endif
+    esp_netif_config_t cfg = {
+        .base = &esp_netif_config,
+        .stack = ESP_NETIF_NETSTACK_DEFAULT_ETH
+    };
+    esp_netif_t *eth_netif = esp_netif_new(&cfg);
+    // Attach Ethernet driver to TCP/IP stack
+    ESP_ERROR_CHECK(esp_netif_attach(eth_netif, esp_eth_new_netif_glue(eth_handle)));
+
+    // Register user defined event handers
+    ESP_ERROR_CHECK(esp_event_handler_register(ETH_EVENT, ESP_EVENT_ANY_ID, &eth_event_handler, NULL));
+    ESP_ERROR_CHECK(esp_event_handler_register(IP_EVENT, IP_EVENT_ETH_GOT_IP, &got_ip_event_handler, NULL));
+
+    ESP_ERROR_CHECK(esp_eth_start(eth_handle));
+
+#if !CONFIG_EXAMPLE_DHCP_SERVER_EN
+    vTaskDelay(pdMS_TO_TICKS(7000));
+    esp_eth_stop(eth_handle);
+    vTaskDelay(pdMS_TO_TICKS(500));
+    esp_eth_start(eth_handle);
```

Comment:
Could you  explain why you're restarting `esp_eth` here?

Reply:
Good catch. It's a leftover from testing. There used to be an issue that Ethernet didn't start when autonegotiation disabled and Ethernet was stopped.